string.rb

```diff
@@ -1,556 +1,556 @@
+# A +String+ object has an arbitrary sequence of bytes,
+# typically representing text or binary data.
+# A +String+ object may be created using String::new or as literals.
 #
-#  String objects differ from Symbol objects in that Symbol objects are
-#  designed to be used as identifiers, instead of text or data.
+# String objects differ from Symbol objects in that Symbol objects are
+# designed to be used as identifiers, instead of text or data.
 #
-#  You can create a +String+ object explicitly with:
+# You can create a +String+ object explicitly with:
 #
-#  - A {string literal}[rdoc-ref:syntax/literals.rdoc@String+Literals].
-#  - A {heredoc literal}[rdoc-ref:syntax/literals.rdoc@Here+Document+Literals].
+# - A {string literal}[rdoc-ref:syntax/literals.rdoc@String+Literals].
+# - A {heredoc literal}[rdoc-ref:syntax/literals.rdoc@Here+Document+Literals].
 #
-#  You can convert certain objects to Strings with:
+# You can convert certain objects to Strings with:
 #
-#  - \Method #String.
+# - \Method #String.
 #
-#  Some +String+ methods modify +self+.
-#  Typically, a method whose name ends with <tt>!</tt> modifies +self+
-#  and returns +self+;
-#  often a similarly named method (without the <tt>!</tt>)
-#  returns a new string.
+# Some +String+ methods modify +self+.
+# Typically, a method whose name ends with <tt>!</tt> modifies +self+
+# and returns +self+;
+# often, a similarly named method (without the <tt>!</tt>)
+# returns a new string.
 #
-#  In general, if there exist both bang and non-bang version of method,
-#  the bang! mutates and the non-bang! does not.
-#  However, a method without a bang can also mutate, such as String#replace.
+# In general, if both bang and non-bang versions of a method exist,
+# the bang method mutates and the non-bang method does not.
+# However, a method without a bang can also mutate, such as String#replace.
 #
-#  == Substitution Methods
+# == Substitution Methods
 #
-#  These methods perform substitutions:
+# These methods perform substitutions:
 #
-#  - String#sub: One substitution (or none); returns a new string.
-#  - String#sub!: One substitution (or none); returns +self+ if any changes,
-#    +nil+ otherwise.
-#  - String#gsub: Zero or more substitutions; returns a new string.
-#  - String#gsub!: Zero or more substitutions; returns +self+ if any changes,
-#    +nil+ otherwise.
+# - String#sub: One substitution (or none); returns a new string.
+# - String#sub!: One substitution (or none); returns +self+ if any changes,
+#   +nil+ otherwise.
+# - String#gsub: Zero or more substitutions; returns a new string.
+# - String#gsub!: Zero or more substitutions; returns +self+ if any changes,
+#   +nil+ otherwise.
 #
-#  Each of these methods takes:
+# Each of these methods takes:
 #
-#  - A first argument, +pattern+ (string or regexp),
-#    that specifies the substring(s) to be replaced.
+# - A first argument, +pattern+ (String or Regexp),
```

Comment:
Capitalizing changes like this will make them link to the relevant classes, which is an improvement IMO.